File: src/main/java/com/ares/http/ArtworkApi.java
package com.ares.http;

import com.ares.entity.Actress;
import com.ares.entity.ActressDetail;
import com.ares.entity.BaseSearchItem;
import com.ares.entity.MovieSearchItem;
import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

import java.util.List;

public interface ArtworkApi {




    @GET("/artwork/actresses/{page}")
    Observable<List<Actress>> getActressList(@Path("page") int page);

    @GET("/artwork/actressInfo")
    Observable<ActressDetail> getActressDetail(@Query("id") String id);

    @GET("/artwork/search/{keyword}/{page}")
    Observable< List<BaseSearchItem>> getSearchList(@Path("keyword") String keyword,@Path("page") int page,@Query(value = "type") int type);


    @GET("/artwork/{code}")
    Observable<MovieSearchItem> searchArtWork(@Path("code")String code);
}

File: src/main/java/com/ares/http/ArtworkApi.kt
package com.ares.http

import com.ares.entity.*
import io.reactivex.Observable
import retrofit2.http.Field
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ArtworkApi {


    @GET("/artwork/actresses/{page}")
    fun getActressList(@Path("page") page: Int): Observable<List<Actress>>

    @GET("/artwork/actressInfo")
    fun getActressDetail(@Query("id") id: String): Observable<ActressDetail>

    @GET("/artwork/search/{keyword}/{page}")
    fun getSearchList(@Path("keyword") keyword: String, @Path("page") page: Int, @Query(value = "type") type: Int): Observable<List<BaseSearchItem>>


    @GET("/artwork/{code}")
    fun searchArtWork(@Path("code") code: String): Observable<MovieSearchItem>

    @GET("/artwork/video/{code}")
    fun searchArtWorkVideo(@Path("code") code: String): Observable<List<VideoSearchItem>>

}
